adala/runtimes/_litellm.py

```diff
@@ -59,6 +84,35 @@ def _format_error_dict(e: Exception) -> dict:
     return error_dct
 
 
+def _log_llm_exception(e) -> dict:
+    dct = _format_error_dict(e)
+    base_error = f"Inference error {dct['_adala_message']}"
+    tb = traceback.format_exc()
+    logger.error(f"{base_error}\nTraceback:\n{tb}")
+    return dct
+
+
+def _update_with_usage(data: Dict, usage: Usage, model: str) -> None:
```

Comment:
nit: can we just return the usage dict and do something like `data.update(_update_with_usage(usage, model)`? I figure that's more transparent (likely would change the name of the function at that point, I guess)